test/python/visualization/test_circuit_text_drawer.py

```diff
@@ -1688,22 +1688,22 @@ def test_control_gate_label_with_cond_2_med(self):
         See https://github.com/Qiskit/qiskit-terra/issues/4361"""
         expected = "\n".join(
             [
-                "        ┌──────┐",
-                "q_0: |0>┤ my h ├",
-                "        └──┬───┘",
-                "q_1: |0>───■────",
-                "         my ch  ",
-                "           ║    ",
-                "   c: 0 ═══■════",
-                "          0x1   ",
+                "          ┌──────┐ ",
+                "q_0: |0>──┤ my h ├─",
+                "          └──┬───┘ ",
+                "q_1: |0>─────■─────",
+                "         my ctrl-h ",
+                "             ║     ",
+                "   c: 0 ═════■═════",
+                "            0x1    ",
```

Comment:
Can we revert this test case?  The behaviour that appears with this PR and this test is probably something we want to make a positive decision on, and potentially test both it and your new form (with the space not aligned).

For clarity: with this PR, the original test produces
```text
        ┌──────┐
q_0: |0>┤ my h ├
        └──┬───┘
q_1: |0>───■────
         my║ch
   c: 0 ═══■════
          0x1
```
instead.  I feel like that (^) behaviour is probably actually suitable for `vertical_compression=medium` - it compresses vertically when there's a "space" so nothing visible gets erased, whereas "high" _always_ compresses and "low" never does.

Perhaps leave the test as-is, and add your new one as a separate case?

Reply:
Ha, I was hoping nobody would notice that. It works visually either way and it's probably a pretty rare case. I'll add the old test back with the new one.